js/packages/mobile-wallet-adapter-protocol-web3js/src/transact.ts

```diff
@@ -50,42 +50,20 @@ export async function transact<TReturn>(
                         case 'signAndSendTransactions':
                             target[p] = async function ({
-                                connection,
-                                fee_payer: feePayer,
                                 transactions,
                                 ...rest
                             }: Parameters<Web3MobileWallet['signAndSendTransactions']>[0]) {
-                                let latestBlockhashPromise: ReturnType<
-                                    InstanceType<typeof Connection>['getLatestBlockhash']
-                                >;
-                                let targetCommitment: Finality;
-                                switch (connection.commitment) {
-                                    case 'confirmed':
-                                    case 'finalized':
-                                    case 'processed':
-                                        targetCommitment = connection.commitment;
-                                        break;
-                                    default:
-                                        targetCommitment = 'finalized';
-                                }
-                                function getLatestBlockhashPromise() {
-                                    if (latestBlockhashPromise == null) {
-                                        latestBlockhashPromise = connection.getLatestBlockhash({
-                                            commitment: targetCommitment,
-                                        });
-                                    }
-                                    return latestBlockhashPromise;
-                                }
                                 const payloads = await Promise.all(
                                     transactions.map(async (transaction) => {
+                                        // Message serialization will fail unless there is a non-null fee payer.
                                         if (transaction.feePayer == null) {
-                                            transaction.feePayer = feePayer;
+                                            transaction.feePayer = PublicKey.default;
                                         }
+                                        // Message serialization will fail unless there is a non-falsy blockhash.
                                         if (transaction.recentBlockhash == null) {
-                                            const { blockhash } = await getLatestBlockhashPromise();
-                                            transaction.recentBlockhash = blockhash;
+                                            transaction.recentBlockhash = '11111111111111111111111111111111';
                                         }
-                                        const serializedTransaction = transaction.serialize({
+                                        const serializedTransaction = await transaction.serialize({
```

Comment:
Despite this, I have no idea how a wallet would know that these are garbage and should be replaced.

Reply:
OK, I think either:

1. This was right before and I should close this PR, or
2. If someone sends a `Transaction` without a `feePayer` or `recentBlockhash` to `mobile-wallet-adapter-protocol-web3js` then we should _let_ it fail.

Thoughts?

Reply:
`feePayer` seems straightforward to set to the expected account

Is the issue with setting `recentBlockhash` that we don't know what commitment level it should have?

Reply:
No, I just want to make `mobile-wallet-adapter-protocol-web3js` dumber. It does way too much. It's meant to be _only_ a wrapper around web3.js datatypes (ie. `mobile-wallet-adapter-protocol` takes a `payloads: Base64EncodedTransaction[]` while `mobile-wallet-adapter-protocol-web3js` takes a `transactions: Transaction[]`).